build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RunInstrumentationTests.cs

```diff
@@ -100,8 +100,10 @@ public override bool Execute ()
 				},
 
 				new CommandInfo {
-					ArgumentsGenerator = () => $"{AdbTarget} {AdbOptions} pull \"{targetTestResultsPath}\" \"{NUnit2TestResultsFile}\"",
-					ShouldRun = () => !String.IsNullOrEmpty (targetTestResultsPath)
+					ArgumentsGenerator = () => $"{AdbTarget} {AdbOptions} shell run-as {PackageName} cat \"{targetTestResultsPath}\"",
```

Comment:
A possible problem with this change is that it requires that the installed `.apk` be "debuggable" (`//application/@android:debuggable`=true), which might not be true for all of our unit tests.  I wonder if this is why some of the on-device tests report "crashes"…. Requires more investigation on my part.

Reply:
In this PR I've been adding the debuggable attribute where we need it , see https://github.com/xamarin/xamarin-android/pull/5257/files#diff-e65c8d18ccd429c3af870ee7c9b878c2f73cc875493baadf86dd75c0fb60d7c7R10. 

I may well have missed a few.

Reply:
I reworked this system so that it only used Debuggable on API 30 as thats the only emulator which needs it.